assembly/src/assembler/procedure_cache.rs

```diff
@@ -42,6 +42,11 @@ impl ProcedureCache {
         self.procedures.get(mast_root)
     }
 
+    /// Returns a MAST root ([RpoDigest]) reference corresponding to the provided [ProcedureId].
+    pub fn get_root_by_id(&self, id: &ProcedureId) -> Option<RpoDigest> {
+        self.proc_id_map.get(id).cloned()
+    }
```

Comment:
minor nit: I'd rename this into `get_proc_root_by_id()` to make it slightly less ambiguous.